core/mutator/mutator.go

```diff
@@ -53,10 +53,14 @@ type Mutator interface {
 }
 
 // Mutation reads and writes mutations to the database.
+// TODO: Add mapID to this interface to support multiple maps per server.
 type Mutation interface {
-	// Read reads all mutations for a specific given mapID, epoch, and index.
-	// TODO(cesarghali): Add mapID to this interface to support multiple maps per server.
-	Read(ctx context.Context, txn transaction.Txn, epoch int64, index []byte) ([]byte, error)
-	// Write saves the mutation in the database.
-	Write(ctx context.Context, txn transaction.Txn, epoch int64, index, mutation []byte) error
+	// Read reads all mutations for a specific given mapID, sequence, and
+	// index.
+	Read(ctx context.Context, txn transaction.Txn, sequence uint64, index []byte) ([]byte, error)
```

Comment:
Clients of this API won't necessarily know what sequence number to use here. All they will know is that the sequence number of the last thing they submitted will be greater than anything else in the DB. 

The sequencer and monitors will need to `Read` a set of mutations from `A` to either `B` or `latest`. 

Consider `ReadRange(txn, mapID, start, count/end)` as an alternative?